src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -0,0 +1,1025 @@
+        isr_handler(NULL, EXTI10);
+        TRACE("gpio_driver: exti10 interrupt triggered\n");
+    } else if (exti_get_flag_status(EXTI11)) {
+        exti_reset_request(EXTI11);
+        isr_handler(NULL, EXTI11);
+        TRACE("gpio_driver: exti11 interrupt triggered\n");
+    } else if (exti_get_flag_status(EXTI12)) {
+        exti_reset_request(EXTI12);
+        isr_handler(NULL, EXTI12);
+        TRACE("gpio_driver: exti12 interrupt triggered\n");
+    } else if (exti_get_flag_status(EXTI13)) {
+        exti_reset_request(EXTI13);
+        isr_handler(NULL, EXTI13);
+        TRACE("gpio_driver: exti13 interrupt triggered\n");
+    } else if (exti_get_flag_status(EXTI14)) {
+        exti_reset_request(EXTI14);
+        isr_handler(NULL, EXTI14);
+        TRACE("gpio_driver: exti14 interrupt triggered\n");
+    } else if (exti_get_flag_status(EXTI15)) {
+        exti_reset_request(EXTI15);
+        isr_handler(NULL, EXTI15);
+        TRACE("gpio_driver: exti15 interrupt triggered\n");
+    } else {
+        AVM_LOGE(TAG, "exti15_10_isr triggered, but no match found!");
+    }
+}
+
+static term gpiodriver_set_level(Context *ctx, term cmd)
+{
+    term gpio_pin_tuple = term_get_tuple_element(cmd, 1);
+    term level = term_get_tuple_element(cmd, 2);
+
+    return gpio_digital_write(ctx, gpio_pin_tuple, level);
+}
+
+static term gpiodriver_set_direction(Context *ctx, term cmd)
+{
+    term gpio_tuple = term_get_tuple_element(cmd, 1);
+    term direction = term_get_tuple_element(cmd, 2);
+
+    return setup_gpio_pin(ctx, gpio_tuple, direction);
+}
+
+static term gpiodriver_read(Context *ctx, term cmd)
+{
+    term gpio_pin_tuple = term_get_tuple_element(cmd, 1);
+    return gpio_digital_read(ctx, gpio_pin_tuple);
+}
+
+static bool gpiodriver_is_gpio_attached(struct GPIOData *gpio_data, term gpio_bank_atom, uint16_t pin)
+{
+    struct ListHead *item;
+    LIST_FOR_EACH (item, &gpio_data->gpio_listeners) {
+        struct GPIOListenerData *gpio_listener = GET_LIST_ENTRY(item, struct GPIOListenerData, gpio_listener_list_head);
+        if ((gpio_listener->gpio_pin == pin) && (gpio_listener->bank_atom == gpio_bank_atom)) {
+            return true;
+        }
+    }
+    return false;
+}
+
+static term gpiodriver_set_int(Context *ctx, int32_t target_pid, term cmd)
+{
+    char *bank_string;
+    char *trigger_string;
+
+    term gpio_tuple = term_get_tuple_element(cmd, 1);
+    if (UNLIKELY(!term_is_tuple(gpio_tuple))) {
+        AVM_LOGE(TAG, "Invalid GPIO Pin tuple, expect {Bank, Pin}.");
+        return ERROR_ATOM;
+    }
+    term gpio_bank_atom = term_get_tuple_element(gpio_tuple, 0);
+    if (UNLIKELY(!term_is_atom(gpio_bank_atom))) {
+        AVM_LOGE(TAG, "Bank parameter of pin tuple must be an atom!");
+        return ERROR_ATOM;
+    }
+    uint32_t gpio_bank = (uint32_t) interop_atom_term_select_int(gpio_bank_table, gpio_bank_atom, ctx->global);
+    bank_string = interop_atom_to_string(ctx, gpio_bank_atom);
```

Comment:
a few more to fix here

Reply:
Yes, I fixed them in one commit, but forgot there were several others in the interrupt commit.

Reply:
fixed